flake.nix

```diff
@@ -15,96 +15,133 @@
+
+          craneLib = (crane.mkLib pkgs).overrideToolchain pkgs.rust-bin.stable.latest.minimal;
+          buildInputs = with pkgs; [
             openssl
-            pkg-config
-            protobuf
             sqlite
             zlib
           ];
 
-          cargoConfig = builtins.fromTOML (builtins.readFile ./.cargo/config.toml); # TODO: Set the target CPU conditionally
-          cargoToml = builtins.fromTOML (builtins.readFile ./Cargo.toml);
+          nativeBuildInputs = with pkgs; [
+            protobuf
+            pkg-config
+            rustPlatform.bindgenHook
+          ];
+
           src = pkgs.lib.cleanSourceWith {
             src = pkgs.lib.cleanSource ./.;
             filter = name: type:
               let baseName = baseNameOf (toString name);
               in !(baseName == "flake.lock" || pkgs.lib.hasSuffix ".nix" baseName);
           };
-          version = cargoToml.workspace.package.version;
 
-          basePackage = {
-            inherit version src;
+          commonArgs = {
+            inherit src stdenv buildInputs nativeBuildInputs;
+
+            strictDeps = true;
 
             meta = {
               description = "ActivityPub-federated microblogging";
               homepage = "https://joinkitsune.org";
             };
 
-            cargoLock = {
-              lockFile = ./Cargo.lock;
-              allowBuiltinFetchGit = true;
-            };
-
-            nativeBuildInputs = baseDependencies;
-
-            PKG_CONFIG_PATH = "${pkgs.openssl.dev}/lib/pkgconfig"; # Not sure why this is broken but it is
-            RUSTFLAGS = builtins.concatStringsSep " " cargoConfig.build.rustflags; # Oh god help.
-
-            checkFlags = [
-              # Depend on creating an HTTP client and that reads from the systems truststore
-              # Because nix is fully isolated, these types of tests fail
-              #
-              # Some (most?) of these also depend on the network? Not good??
-              "--skip=activitypub::fetcher::test::federation_allow"
-              "--skip=activitypub::fetcher::test::federation_deny"
-              "--skip=activitypub::fetcher::test::fetch_actor"
-              "--skip=activitypub::fetcher::test::fetch_note"
-              "--skip=resolve::post::test::parse_mentions"
-              "--skip=webfinger::test::fetch_qarnax_ap_id"
-              "--skip=basic_request"
-              "--skip=json_request"
-            ];
+            OPENSSL_NO_VENDOR = 1;
+            NIX_OUTPATH_USED_AS_RANDOM_SEED = "aaaaaaaaaa";
           };
+
+          cargoTestExtraArgs = pkgs.lib.strings.concatStringsSep " " [
+            "--"
+            # Depend on creating an HTTP client and that reads from the systems truststore
+            # Because nix is fully isolated, these types of tests fail
+            #
+            # Some (most?) of these also depend on the network? Not good??
+            "--skip=activitypub::fetcher::test::federation_allow"
```

Comment:
Since we don't run tests anymore, can we remove these arguments?

Reply:
Good point! Done!